src/cursor/abstract_cursor.ts

```diff
@@ -295,9 +294,9 @@ export abstract class AbstractCursor<
   }
 
   /** Get the next available document from the cursor, returns null if no more documents are available. */
-  next<T = TSchema>(): Promise<T | null>;
-  next<T = TSchema>(callback: Callback<T | null>): void;
-  next<T = TSchema>(callback?: Callback<T | null>): Promise<T | null> | void {
+  next(): Promise<TSchema | null>;
+  next(callback: Callback<TSchema | null>): void;
```

Comment:
Hi! I'm a bit confused as to the purpose of the explicit `| null` being used?

Reply:
Our next functions explicitly return null when the end of the cursor is reached. Our callback definition only lets the second are be potentially undefined, so the `| null` is very technically necessary, if one is triple equals checking for null.

Reply:
Makes sense. The following is purely pedantry and feel free to ignore it, because I likely would if I were you lol.

For transparency, do think a helper type here would be useful? So instead of showing the | null it would be `Promise<EndOfCursor<TSchema>>` where EndOfCursor is defined as `T | null`? Again, super minute and what not. Marking this comment as resolved so it's out of mind lol. Thanks for answering my questions!

Reply:
It's true we have this repeated in a number of places, consolidating it to type would help us make sure we don't get one override different from the others, but I'll defer that to an enhancement effort down the line for now I want this PR to unblock the issue with the generics, but I do appreciate the suggestion! thanks!